src/infrastructure/repositories/CrudRepository.ts

```diff
@@ -165,33 +165,36 @@ export class CrudRepository<TModel> implements ICrudRepository<TModel>, OnModule
      * @param model
      * @param transactionHandler
      */
-    async save(model: TModel, transactionHandler?: (callback) => Promise<void>): Promise<TModel> {
-        const saver = async (manager: EntityManager, nextModel: TModel) => {
+    async save(
+        model: TModel,
+        transactionHandler?: TransactionHandler<TModel>,
+    ): Promise<TModel> {
+        const saver = async (manager: EntityManager, nextModel: TModel): Promise<TModel> => {
             const entity = this.modelToEntity(nextModel);
             const newEntity = await manager.save(entity);
-            DataMapper.applyValues(nextModel, newEntity, TRANSFORM_TYPE_FROM_DB);
+            return this.entityToModel(newEntity);
         };
 
         if (transactionHandler) {
-            await this.dbRepository.manager.transaction(async (manager) => {
-                await transactionHandler(async () => {
-                    await this.saveInternal({save: (nextModel) => saver(manager, nextModel)}, model);
+            return this.dbRepository.manager.transaction<TModel>(async (manager) => {
+                return transactionHandler(async () => {
+                    return this.saveInternal({
+                        save: (nextModel) => saver(manager, nextModel),
+                    }, model);
```

Comment:
Раз уж пошли изменения в отступах, давайте явно покажем что функция 2 параметра принимает:
```suggestion
                    return this.saveInternal(
                        {save: (nextModel) => saver(manager, nextModel)},
                        model,
                    );
```